### PR TITLE
RAC-8: add quantified association field

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

- Adds a new column in the product / product model DB tables
- Adds a new field in the product / product model entities named "quantified associations"
- Adds migration

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | Ok
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by tseho** [comment]
What is the default value of the column after the ALTER TABLE ?
If it's null, does doctrine correctly initialize an array in `protected $rawQuantifiedAssociations;` ?

**Comment by SamirBoulil** [comment]
> What is the default value of the column after the ALTER TABLE ?
It is null

> If it's null, does doctrine initialize an array in `protected $rawQuantifiedAssociations;` ?
It does not really matter, we'll write a custom hydrator that will handle null values.

> If it does not, do we have to check if the property is null ? If so, we should mark it as nullable.
We will use NULL Object pattern, we will always hydrate an object.

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php [hunk 1]
```diff
@@ -27,6 +27,9 @@ abstract class AbstractProduct implements ProductInterface
     /** @var array */
     protected $rawValues;
 
+    /** @var array */
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php:30]:
```suggestion
    /** @var array|null */
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/ProductModel.php [hunk 2]
```diff
@@ -25,6 +25,9 @@ class ProductModel implements ProductModelInterface
     /** @var array|object */
     protected $rawValues;
 
+    /** @var array */
```
Comment (tseho) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/ProductModel.php:28]:
```suggestion
    /** @var array|null */
```

**Pull request merged**